velox/type/tests/TimestampTest.cpp

```diff
@@ -299,37 +299,152 @@ TEST(TimestampTest, outOfRange) {
       t.toTimezone(*timezone), "Timestamp is outside of supported range");
 }
 
-TEST(TimestampTest, epochToUtc) {
-  std::tm tm;
-  ASSERT_FALSE(epochToUtc(-(1ll << 60), tm));
-  ASSERT_FALSE(epochToUtc(1ll << 60, tm));
+void checkTm(const std::tm& actual, const std::tm& expected) {
+  ASSERT_EQ(expected.tm_year, actual.tm_year);
+  ASSERT_EQ(expected.tm_yday, actual.tm_yday);
+  ASSERT_EQ(expected.tm_mon, actual.tm_mon);
+  ASSERT_EQ(expected.tm_mday, actual.tm_mday);
+  ASSERT_EQ(expected.tm_wday, actual.tm_wday);
+  ASSERT_EQ(expected.tm_hour, actual.tm_hour);
+  ASSERT_EQ(expected.tm_min, actual.tm_min);
+  ASSERT_EQ(expected.tm_sec, actual.tm_sec);
 }
 
-TEST(TimestampTest, randomEpochToUtc) {
+std::string tmToString(
+    const std::tm& tmValue,
+    const std::string& format,
+    int32_t nanos,
+    const TimestampToStringOptions& options) {
+  auto width = static_cast<int>(options.precision);
+  auto value =
+      options.precision == TimestampToStringOptions::Precision::kMilliseconds
+      ? nanos / 1'000'000
+      : nanos;
+
+  std::ostringstream oss;
+  oss << std::put_time(&tmValue, format.c_str());
+
+  if (options.mode != TimestampToStringOptions::Mode::kDateOnly) {
+    oss << '.' << std::setfill('0') << std::setw(width) << value;
+  }
+
+  return oss.str();
+}
+
+TEST(TimestampUtilTest, epochToUtc) {
+  std::tm tm{};
+  ASSERT_FALSE(Timestamp::epochToUtc(-(1ll << 60), tm));
+  ASSERT_FALSE(Timestamp::epochToUtc(1ll << 60, tm));
+}
+
+TEST(TimestampUtilTest, randomEpochToUtc) {
   uint64_t seed = 42;
```

Comment:
What is the significance of this number ? Why is not the previous randomly generated number ?

Reply:
It doesn't have any special meaning, just randomly chosen.